pkg/apis/openebs/v1alpha1/blockdeviceclaim_types.go

```diff
@@ -39,6 +39,10 @@ type BlockDeviceClaim struct {
 
 // DeviceClaimSpec defines the request details for a BlockDevice
 type DeviceClaimSpec struct {
+
+	// Selector is used to find block devices to be considered for claiming
+	Selector *metav1.LabelSelector `json:"selector,omitempty"`
```

Comment:
Is this the only way NDM is going to select?
It might be good if specs can support others selections in future.

Why not below specs?
```yaml
spec:
  selector:
    labels:
```

Reply:
`LabelSelector` contains both, `MatchLabels` which is a `map[string]string` and `MatchExpressions` which contains `key, operator, and []value`. This is the same k8s label selector.

Reply:
This is the only way by which NDM selects from etcd. i.e this selectors are applied before the actual `List()` call. All other selectors and filters like deviceType / capacity / volumeMode etc will be used on the list of items. For that we are using separate methods available [here](https://github.com/openebs/node-disk-manager/blob/master/pkg/select/blockdevice/filters.go)

Reply:
Are those filters exposed for users to take part in selection.

My point is about selector which is coupled to labels only. In other words we are also saying NDM won't select by annotations ever even in future.

Reply:
yes, they are exposed via other fields in the spec. eg: `VolumeMode`, `ResourceRequirements`

This is the only field for label based selection. I think annotations are a way to store data that is not user facing, but for internal use. so currently we haven't thought of selecting based on annotations. In current setup, we use annotations only to disable reconciliation of resources. cc: @kmova

Reply:
ok.